### PR TITLE
Machine doc moved

### PR DESCRIPTION
**Pull request by deadprogram**
Basically #47 but moved under "Microcontrollers" and with links from each board to its machine package docs.

### CODE REVIEW

File: doc-gen/Makefile [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+all:
+	go run . $(DEVICES)
+
+
+#all:
+#	go run . pca10040 > ../content/doc/pca10040/machine.md
+#	go run . hifive1b > ../content/doc/hifive1b/machine.md
+#	go run . bluepill > ../content/doc/bluepill/machine.md
+#	go run . gameboy-advance > ../content/doc/gameboy-advance/machine.md
+#
```
Comment (aykevl) [inline doc-gen/Makefile]: Oops looks like this is some old code that should really be removed.

File: doc-gen/Makefile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    metro-m4-airlift \
+    circuitplay-express \
+    stm32f4disco \
+    circuitplay-express \
+    gameboy-advance \
+    itsybitsy-m4 \
+    nucleo-f103rb \
+    hifive1b \
+    pinetime-devkit0 \
+    x9pro
```
Comment (aykevl) [inline doc-gen/Makefile]: I see you reordered some of these devices, any reason for that?
The order doesn't really matter, but I kept it at the same order as the smoke tests in the Makefile, for easier updating (maybe?).

Comment (deadprogram) [reply doc-gen/Makefile]: I was going to put it all into alpha order, but instead I just added the missing ones in semi-random order.

File: doc-gen/Makefile (hunk 1 above)
Comment (deadprogram) [reply doc-gen/Makefile]: Yeah, I was wondering. :scissors:

**Pull request merged**